google/cloud/testing_util/validate_metadata.h

```diff
@@ -15,20 +15,60 @@
 #ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_VALIDATE_METADATA_H
 #define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_VALIDATE_METADATA_H
 
-#include "google/cloud/status.h"
+#include "google/cloud/internal/absl_str_replace_quiet.h"
 #include "google/cloud/version.h"
 #include "absl/types/optional.h"
+#include <google/api/annotations.pb.h>
+#include <google/protobuf/descriptor.h>
 #include <gmock/gmock.h>
 #include <grpcpp/generic/async_generic_service.h>
 #include <grpcpp/grpcpp.h>
 #include <map>
+#include <regex>
 #include <string>
 
 namespace google {
 namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace testing_util {
 
+/**
+ * We do not use `::testing::MatchesRegex` because our Windows builds use a
+ * googletest built with `GTEST_USES_SIMPLE_RE`, instead of
+ * `GTEST_USES_POSIX_RE`.
+ */
+MATCHER_P(MatchesMetadataRegex, pattern,
+          "matches the pattern: \"" + pattern + "\"") {
+  // Translate the pattern into a form that can be used by std::regex
+  std::regex regex(absl::StrReplaceAll(pattern, {{"*", "[^/]+"}}));
```

Comment:
The inputs strings are "globs" (hence the need for the replacement), so I would avoid the terms "Regex" in the matcher name and "pattern" in its argument.

And then, it becomes clearer that the stuff about Windows is an implementation detail, rather than something that should be documented about the matcher.

Reply:
Thanks. I was confused about how to refer to these things. I will use your lingo.